internal/service/base/resource_rate_limit_configuration.go

```diff
@@ -0,0 +1,355 @@
+		sdk.DefaultCreateReadRetryable,
+		&response,
+	)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Remove from state if resource is not found
+	if response == nil {
+		resp.State.RemoveResource(ctx)
+		return
+	}
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(data.toState(response)...)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *RateLimitConfigurationResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	// Rate limit configurations are immutable - all fields require replacement
+	// This method should never be called as all fields have RequiresReplace plan modifiers
+}
+
+func (r *RateLimitConfigurationResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var data *RateLimitConfigurationResourceModel
+
+	if r.Client == nil || r.Client.ManagementAPIClient == nil {
+		resp.Diagnostics.AddError(
+			"Client not initialized",
+			"Expected the PingOne client, got nil.  Please report this issue to the provider maintainers.")
+		return
+	}
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Run the API call
+	resp.Diagnostics.Append(legacysdk.ParseResponse(
+		ctx,
+
+		func() (any, *http.Response, error) {
+			fR, fErr := r.Client.ManagementAPIClient.RateLimitingApi.DeleteRateLimitConfiguration(ctx, data.EnvironmentId.ValueString(), data.Id.ValueString()).Execute()
+			return legacysdk.CheckEnvironmentExistsOnPermissionsError(ctx, r.Client.ManagementAPIClient, data.EnvironmentId.ValueString(), nil, fR, fErr)
+		},
+		"DeleteRateLimitConfiguration",
+		legacysdk.CustomErrorResourceNotFoundWarning,
+		sdk.DefaultCreateReadRetryable,
+		nil,
+	)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (r *RateLimitConfigurationResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+
+	idComponents := []framework.ImportComponent{
+		{
+			Label:  "environment_id",
+			Regexp: verify.P1ResourceIDRegexp,
+		},
+		{
+			Label:     "rate_limit_configuration_id",
+			Regexp:    verify.P1ResourceIDRegexp,
+			PrimaryID: true,
+		},
+	}
+
+	attributes, err := framework.ParseImportID(req.ID, idComponents...)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unexpected Import Identifier",
+			err.Error(),
+		)
+		return
+	}
+
+	for _, idComponent := range idComponents {
+		pathKey := idComponent.Label
+
+		if idComponent.PrimaryID {
+			pathKey = "id"
+		}
+
+		resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root(pathKey), attributes[idComponent.Label])...)
+	}
+}
+
+func (p *RateLimitConfigurationResourceModel) expand() *management.RateLimitConfiguration {
+	typeEnum := management.ENUMRATELIMITCONFIGURATIONTYPE_WHITELIST
+
+	data := management.NewRateLimitConfiguration(
+		typeEnum,
```

Comment:
In case of new types being added in the future, we should be reading this from the model rather than hard-coding it here, since it'll come through from the default anyways